subprojects/docs/src/docs/userguide/authoring-builds/build_lifecycle.adoc

```diff
@@ -138,50 +159,64 @@ include::{snippetsPath}/buildlifecycle/buildProjectEvaluateEvents/tests-groovy/b
 include::{snippetsPath}/buildlifecycle/buildProjectEvaluateEvents/tests-kotlin/buildProjectEvaluateEvents.kotlin.out[]
 ----
 
-You can also add a link:{javadocPath}/org/gradle/api/ProjectEvaluationListener.html[ProjectEvaluationListener] to the link:{groovyDslPath}/org.gradle.api.invocation.Gradle.html[Gradle] to receive these events.
+To receive these events via a listener instead of a closure, add a link:{javadocPath}/org/gradle/api/ProjectEvaluationListener.html[ProjectEvaluationListener] to a build's link:{groovyDslPath}/org.gradle.api.invocation.Gradle.html[Gradle] instance.
+
+=== Task Execution Graph Assembly
+
+During project evaluation, Gradle assembles the *task execution graph*: a http://en.wikipedia.org/wiki/Directed_acyclic_graph[DAG] representing the dependency relationships between tasks.
+
+[[sec:task_execution_graph_ready]]
+=== React to Task Execution Graph Assembly
+
+You can receive a notification immediately after Gradle finishes populating a project's task execution graph.
+
+To receive these events, add a link:{javadocPath}/org/gradle/api/execution/TaskExecutionGraphListener.html[TaskExecutionGraphListener] to a project's link:{javadocPath}/org/gradle/api/execution/TaskExecutionGraph.html[TaskExecutionGraph].
+
+=== Task Creation
+
+During project evaluation, Gradle creates tasks and configures inputs, outputs, and actions for those tasks.
```

Comment:
```suggestion
During project evaluation, Gradle registers tasks and their configuration actions. The configuration actions define inputs, outputs, and actions for those tasks. They are evaluated if the task is part of the task graph for the requested tasks.
```

Reply:
Brilliant, good call on changing "creates" to "registers" -- this is one of those subtle nudges we can provide in the docs to get users to use newer, better APIs.